src/Components/EnvironmentPicker/EnvironmentPickerManager.ts

```diff
@@ -13,10 +13,15 @@ import {
 export const getEnvironmentDisplayText = (env: string | IAzureResource) => {
     try {
         if (env) {
-            const urlObj = new URL(
-                getResourceUrl(env, AzureResourceTypes.DigitalTwinInstance)
-            );
-            return urlObj.hostname.split('.')[0];
+            if (typeof env === 'string') {
+                if (new URL(env)) {
+                    return env.split('.')[0].split('://')[1]; // to respect casing in the name of the instance
```

Comment:
I'm curious, why can't we use the object that gets created from the URL class now?

Reply:
URL class is lowercase all the letters in domain name